docs/DeveloperGuide.md

```diff
@@ -1,34 +1,579 @@
+
+**Step 3**. The user executes `allocate` to let the application allocate time slots to `Task`s without a fixed time slot.
+
+`TimeAllocator` iterates through the `TaskList`, and checks whether a `Task` has a fixed time slot or should be allocated by the application to an empty slot. Tasks that have a fixed slot are added to a `fixedTaskList`, while the rest are added to a `flexibleTaskList`. The `Task`s in the `flexibleTaskList` are sorted based on their `importance` and `deadline`. Finally, the `TimeAllocator` iterates through the `fixedTaskList` to find empty time slots in a day, which is then allocated to the `Task`s in the sorted `flexibleTaskList`.
+
+The following sequence diagram illustrates how the allocate operation works:
+
+![TimeAllocatorSequenceDiagram](sequenceDiagrams/TimeAllocator%20diagram.png)
+
+**Step 4.** The user executes `list` to get an overview of the week. The user sees all 5 tasks in the printed timetable.
+
+### Data storage
+The storage mechanism is facilitated by `Storage`. It reads and writes the `Task`s in a `TaskList` to `data.csv`, a csv file located next to the application jar file.
+It implements the following operations:
+
+* `Storage#saveTaskListData` - Writes the current tasks into the save file.
+* `Storage#loadTaskListData` - Loads the tasks in the save file into the application.
+
+These operations are called by the `LogicManager`.
+
+Given below is an example usage scenario and how the storage mechanism behaves at each step.
+
+**Step 1.** The user launches the application for the first time. The `TaskList` is initialized to be empty. At this time, there is no *data.csv* file present. So, when `Storage` calls `Storage#loadTaskListData`, this is detected and an empty *data.csv* file is created next to the jar file. Since there was no save file, the `TaskList` remains empty.
+
+**Step 2.** The user adds a task to the application, by executing `add n/Assignment1 t/1100 D/16-09-2020 d/2 r/Today i/high a/Refer to lecture notes`. The `TaskList` now contains 1 task (Assignment 1). After the command is executed, `LogicManager` calls `Storage#saveTaskListData` to automatically save the tasks in the `TaskList` into the save file.
+
+**Step 3.** The user closes the application. Nothing happens since the data in the `TaskList` is already saved.
+
+**Step 4.** The user launches the application again. The `TaskList` is initialized to be empty. `Storage#loadTaskListData` will read from `data.csv` and add the tasks inside the file into the empty `TaskList`. The `TaskList` now contains the task added earlier (Assignment 1) in **step 3**.
+
+**Step 5.** The user executes `list` to get an overview of the week. The user sees *Assignment 1* in the printed timetable.
+
+The following sequence diagram illustrates how the loading from storage operation works:
+
+![LoadStorageSequenceDiagram](sequenceDiagrams/loadStorage%20diagram.png)
```

Comment:
Since TaskList is not used after it returns to storage after the markTaskAsDone() call, wouldn't it make more sense to stop the activation bar right there?